internal/cache/clockpro.go

```diff
@@ -398,13 +398,41 @@ func (c *shard) metaEvict(e *entry) {
 }
 
 func (c *shard) evict() {
-	for c.targetSize() <= c.sizeHot+c.sizeCold && c.handCold != nil {
-		c.runHandCold()
+	c.reclaim()
+	c.balance()
+}
+
+// Reclaim cache space by evicting cold entries. If the size of cold entries is
+// insufficient, then it continues with pushing handHot to demote hot entries
+// to cold entries.
+func (c *shard) reclaim() {
+	for c.targetSize() <= c.sizeHot+c.sizeCold {
+		if c.sizeCold > 0 {
+			c.runHandCold()
+		} else {
+			c.runHandHot()
```

Comment:
Only moving handHot when there are no cold entries in the list. All cold entries are located between handCold and handHot, so handCold will not go past handHot.